### PR TITLE
Chance dev

### PR DESCRIPTION
**Pull request by chancehaycock**

### CODE REVIEW

File: region.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+        self.label = label
+
+    def num_days(self):
+        return (self.t_max - self.t_min).days
+
+    def num_hours(self):
+        return (self.t_max - self.t_min).days * 24
+
+
+def convert_dates(df: pd.DataFrame) -> pd.DataFrame:
```
Comment (TeddyTW) [inline region.py:36]: is this just to sort out the issues we have been having with datetime format?

File: region.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+#    assert set(['baseline_count', 'actual_count']) <= set(data.columns)
+
+    region_mask = (data['lon'].between(S.x_min, S.x_max)) &\
+                  (data['lat'].between(S.y_min, S.y_max)) &\
+                  (data['measurement_end_utc'] > S.t_min) &\
+                  (data['measurement_end_utc'] <= S.t_max)
+    S_df = data.loc[region_mask]
+    if S_df.empty:
+        return [0, 0]
+#    return S_df['baseline_count'].sum(), S_df['actual_count'].sum()
```
Comment (TeddyTW) [inline region.py:64]: need to consider how this commented out line will work, will the dataframe we feed into this function already have baseline calcualted? or will is be calculated during the function

File: region.py (hunk 1 above)
Comment (chancehaycock) [reply region.py:36]: Yeah - might not need it

File: region.py (hunk 2 above)
Comment (chancehaycock) [reply region.py:64]: Hopefully, your dataframe will have all the detectors in a given location spread over the W days of interest, with each row having a baseline estimate and an actual count. Then this function easily feeds into it

File: region.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
+"""Module to contain all region and grid-based construction code."""
+
+from typing import Type
+import pandas as pd
+from datetime import datetime
+
+class Region:
+    """Class to represent space-time region"""
+
+    def __init__(self, x_min: float, x_max: float, y_min: float, y_max: float,
```
[the remark this answers is not in the document]
Comment (chancehaycock) [reply region.py:11]: Possibly - although in this case (as we have to loop over pretty much all combos, its nice to have them as explicit members

[the remark this answers is not in the document]
Comment (TeddyTW) [reply region.py:11]: sure, didn't think about it that way

File: region.py (hunk 2 above)
Comment (TeddyTW) [reply region.py:64]: okay, so the input to this function is more the output of one of the time series functions?

**Pull request merged**